Dockerfile

```diff
@@ -1,5 +1,19 @@
-# NB: We now cross-compile the go binary locally using the Makefile
-FROM alpine:latest
+FROM golang:1.20-alpine as builder
```

Comment:
We run CI on `stable` and `oldstable`, which now map to 1.21 and 1.20, correct? This means that these will point to 1.22 and 1.21 come Feb? I'm just worried that this 1.20 ref will become stale.

So my suggestions: 1. Bump this to 1.21 since it's the most up to date, 2. Move away from `stable`/`oldstable`.

Reply:
i vote bumping to 1.21 in the dockerfile, the stable/oldstable thing is from dx and they prefer that after being burned by gh actions in the past.

Reply:
Burned in what way? I'm worried _we'll_ be burned here as we run the risk of CI running on a different platform than our release pipeline.

Reply:
@lgarber-akamai can elaborate but it sounded to me like doing stable/oldstable helped dx find go problems faster during the PR process instead of waiting last minute. i guess it's not being "burned by gh actions" so much as gh inaction?

if i had to choose my personal preference is the dockerfile is pinned to the go.mod version and the `make build` job has stable/oldstable... this pins the release artifact to the project's go version and allows the PR process to always have latest go to find problems faster.

Reply:
Sorry for the late follow-up on this 🙂 

Looking back it'd probably make sense to explicitly pin the CI, Dockerfile, and go.mod to go 1.21 (or 1.20) to keep things consistent.